### ISSUE TITLE
Built-in jumping/falling lookahead for 2D?

### ISSUE THREAD

**Issue by elvisish** [ramokz/phantom-camera#228]
### Project Type

2D

### Feature Description

Is it possible to have a built-in lookahead distance for jumping and falling that adjusts depending on the falling and jumping speed? I believe it can be done currently by manually adjusting the Follow Target Offsets depending on the Y velocity, but it would be very useful to have it out-of-the-box for a lot of 2D games.

### Use Cases

If the character is falling pretty fast or jumping/springing quickly, this would ensure the player doesn't leave the screen.

### Importance

Low - there are workarounds where the proposed feature would just simplify things

### Usage

Often - a significant amount of projects can find this useful

**Comment by ramokz** [comment]
It's not a built-in feature at the minute, but there are plans for implementing it.

There is an existing PR for that, with [an explanation](https://github.com/ramokz/phantom-camera/pull/160#issuecomment-1872613344) for how it ought to work.
TLDR; the velocity of the follow target would offset the camera in the direction the target is moving.

**Comment by elvisish** [comment]
> the velocity of the follow target would offset the camera in the direction the target is moving.

Yep, it should also maintain the same distance depending on velocity (so super fast falling, jumping or running doesn't allow the tweening camera position to lag behind).

**Comment by elvisish** [comment]
Is look-ahead still planned? The original PR seems to be closed/deleted: https://github.com/ramokz/phantom-camera/pull/160#issuecomment-1872613344

**Comment by ramokz** [comment]
Short answer, support is still planned.

That PR you linked to was closed as it was based on a hardcoded lookahead value, whereas the behavior should ideally be dynamic based on the target's velocity instead; with some parameters around that.

Another PR (#284) also aimed at implementing it, but that PR has, at this point, likely been abandoned by the author.

Am personally caught up with more fundamental changes / improvements to the addon, so it's not at the top of my todo list atm.

**Comment by elvisish** [comment]
> Short answer, support is still planned.

Great, I think it's also important to note that look ahead should be factored into framed mode, currently if you're using framed, the player is offset to the center of the screen by the framed dead zone amount, which makes it difficult to see enemies for the sake of having the camera move a bit later. The same goes for slow lerp speeds. Look ahead should always be the last thing calculated so it can be used to effectively recenter the player (or even offset them negatively from the center for even better visibility ahead).

**Comment by Mustfa-IT** [comment]
@elvisish @ramokz I’ve submitted a PR implementing this feature request: #612.
Would appreciate any feedback or testing when you have time.

**Comment by ramokz** [comment]
Resolved in #612
